### PR TITLE
5.1 & 5.2

### PR DESCRIPTION
**Pull request by Fulgore2002**

### CODE REVIEW

File: Assets/Prefabs/Bad 1.prefab
Comment (dantogno) [inline Assets/Prefabs/Bad 1.prefab:1]: What is bad 1? (answer was, an object that if you click it, it takes away points). 

In light of that, I would a better name would be like hazardObject or penaltyObject, etc.

File: Assets/Scripts/Target.cs [hunk 1]
```diff
@@ -12,9 +12,14 @@ public class Target : MonoBehaviour
     private float _maxTorque = 10;
```
Comment (xc-a11y) [inline Assets/Scripts/Target.cs:12]: how would you explain these functions?

try adding a summary to explain the functions, or tooltips.

File: Assets/Scripts/Target.cs [hunk 2]
```diff
@@ -12,9 +12,14 @@ public class Target : MonoBehaviour
     private float _maxTorque = 10; 
     private float _xRange = 4; 
     private float _ySpawnPos = -2;
+    public ParticleSystem _explosionParticle;
     public int _pointValue;
 
-    // Start is called before the first frame update
+    /// <summary> 
+    /// Initializes the target object's Rigidbody and GameManager component, 
+    /// and applies random force and torque to the Rigidbody.
```
Comment (xc-a11y) [inline Assets/Scripts/Target.cs:20]: good use of summary!

File: Assets/Scripts/GameManager.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using UnityEngine.SocialPlatforms.Impl;
+
+public class GameManager : MonoBehaviour
+{
+    public List<GameObject> targets;
```
Comment (dantogno) [inline Assets/Scripts/GameManager.cs:9]: Do other classes access these public fields?

File: Assets/Scripts/Target.cs [hunk 4]
```diff
@@ -25,18 +30,21 @@ void Start()
         transform.position = RandomSpawnPos();
```
Comment (xc-a11y) [inline Assets/Scripts/Target.cs:30]: remove extra line on line 27 here.

File: Assets/Scripts/GameManager.cs (hunk 3 above)
Comment (dantogno) [reply Assets/Scripts/GameManager.cs:9]: If other classes don't need to access these, we should private and expose them to the editor with `[SerializeField]`

File: Assets/Scripts/GameManager.cs [hunk 5]
```diff
@@ -19,11 +19,6 @@ void Start()
         UpdateScore(0);
```
Comment (xc-a11y) [inline Assets/Scripts/GameManager.cs:19]: remove "_score = 0;"

File: Assets/Scripts/GameManager.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using UnityEngine.SocialPlatforms.Impl;
+
+public class GameManager : MonoBehaviour
+{
+    public List<GameObject> targets;
+    public TextMeshProUGUI _scoreText;
```
Comment (dantogno) [inline Assets/Scripts/GameManager.cs:10]: awesome to see using naming convention. If this needs to be public, you would use PascalCase `ScoreText`, but I think this probably should be private and exposed with SerializeField, which means the current name is great.

File: Assets/Scripts/GameManager.cs [hunk 7]
```diff
@@ -34,6 +29,10 @@ IEnumerator SpawnTarget()
         }
     }
 
+    /// <summary> 
+    /// Updates the current score by adding the specified value and updates the score display text.
```
Comment (xc-a11y) [inline Assets/Scripts/GameManager.cs:33]: nice summary here!

File: Assets/Scripts/GameManager.cs [hunk 8]
```diff
@@ -34,6 +29,10 @@ IEnumerator SpawnTarget()
         }
     }
 
+    /// <summary> 
+    /// Updates the current score by adding the specified value and updates the score display text. 
+    /// </summary>
+    /// <param name="scoreToAdd">
```
Comment (xc-a11y) [inline Assets/Scripts/GameManager.cs:35]: what is this for?

File: Assets/Scripts/Target.cs [hunk 9]
```diff
@@ -46,10 +54,12 @@ Vector3 RandomForce()
     {
```
Comment (xc-a11y) [inline Assets/Scripts/Target.cs:54]: what do these last things do?

try adding a short description to it with a summary or comment.

**Comment by xc-a11y** [comment]
Overall, pretty good but is missing some minor conventional things that can easily be fixed.

File: Assets/Scripts/GameManager.cs [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            yield return new WaitForSeconds(_spawnRate); 
+            int index = Random.Range(0, targets.Count); 
+            Instantiate(targets[index]);
+        }
+    }
+
+    /// <summary> 
+    /// Updates the current score by adding the specified value and updates the score display text. 
+    /// </summary>
```
Comment (dantogno) [inline Assets/Scripts/GameManager.cs:34]: Great work adding summary of public function!

File: Assets/Scripts/Target.cs [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+public class Target : MonoBehaviour
+{
+    private Rigidbody _targetRb;
+    private GameManager _gameManager;
+    private float _minSpeed = 12; 
+    private float _maxSpeed = 16; 
+    private float _maxTorque = 10; 
+    private float _xRange = 4; 
+    private float _ySpawnPos = -2;
+    public ParticleSystem _explosionParticle;
```
Comment (dantogno) [inline Assets/Scripts/Target.cs:15]: unless another class needs access to _explosionParticle you should use SerializeField to expose it and make it private.

**Review by dantogno — APPROVED** [review]
Great work using our class style guide!

**Pull request merged**